packages/slider/src/useSlider.ts

```diff
@@ -0,0 +1,364 @@
+    if (trackElement.current === null) {
+      return;
+    }
+
+    // we need to check the scope/runtime and make sure it has resize observer
+    // we should also evaluate if a library may be a better solution than using a "newish" browser API
+    // lastly - should we be handling this in the container hook? Or could we get the info per event?
+    if ('ResizeObserver' in window) {
+      // Store track element ref in a variable, per eslint's react-hooks/exhaustive-deps warning, which says:
+      // “The ref value 'trackElement.current' will likely have changed by the time this effect cleanup function runs.
+      // If this ref points to a node rendered by React, copy 'trackElement.current' to a variable inside the effect, and use that variable in the cleanup function.”
+      const initialTrackElement = trackElement.current;
+
+      const resizeObserver = new ResizeObserver(
+        throttle(
+          () => {
+            setTrackElementDimensions(
+              (initialTrackElement as HTMLElement)!.getBoundingClientRect()
+            );
+          },
+          1000,
+          { leading: false }
+        )
+      );
+
+      resizeObserver.observe(initialTrackElement);
+
+      // eslint-disable-next-line consistent-return
+      return () => {
+        resizeObserver.unobserve(initialTrackElement);
+      };
+    }
+  }, [trackElement, setTrackElementDimensions]);
+
+  /**
+   * @todo Accommodate vertical orientation
+   */
+  const getValueClosestToMouse = useCallback(
+    (coordinate: number) => {
+      const mouseDistanceFromLeftEdge = coordinate - trackElementDimensions!.left;
+      const mouseTargetPosition = trackElementDimensions!.width / mouseDistanceFromLeftEdge;
+
+      let value = Math.round(max / mouseTargetPosition);
+
+      if (rtl === true) {
+        value = max - value;
+      }
+
+      return value;
+    },
+    [trackElementDimensions, rtl, max]
+  );
+
+  /**
+   * @todo Accommodate vertical orientation
+   */
+  const handleTrackClick = useCallback(
+    (event: MouseEvent) => {
+      if (isInteractive) {
+        const closestRangeValue = getValueClosestToMouse(event.clientX);
+        const closestThumbValue = state.reduce((previousValue: number, currentValue: number) => {
+          return Math.abs(currentValue - closestRangeValue) <
+            Math.abs(previousValue - closestRangeValue)
+            ? currentValue
+            : previousValue;
+        });
+        const closestThumbIndex = state.findIndex(
+          (element: number) => element === closestThumbValue
+        );
+
+        dispatch(
+          setThumbValue({
+            index: closestThumbIndex,
+            value: closestRangeValue,
+            min,
+            max
+          })
+        );
+      }
+    },
+    [isInteractive, getValueClosestToMouse, state, min, max]
+  );
+
+  /**
+   * @todo Accommodate vertical orientation
+   * @todo move to document mouse move (using environment) instead of track
```

Comment:
Hey @Francois-Esquire, something to talk about during our pairing session tomorrow: I thought the document mouse move might be in addition to the track movement, rather than instead of?

Reply:
@jgorfine-zendesk I think we may have to handle this in the Slider container hook due to the state being internal (uncontrolled by default).